### ISSUE TITLE
[3.0.0] Volt: Macro does not work!

### ISSUE THREAD

**Issue by xerron** [phalcon/cphalcon#12118]
- Phalcon Micro
- View/Simple

service view:

``` php

use Phalcon\Mvc\View\Engine\Volt as VoltEngine;
use Phalcon\Mvc\View\Simple as View;

$di->setShared('view', function () {    

 $config = $this->getConfig();
 $view = new View();
 $view->setViewsDir($config->application->viewsDir);
 $view->registerEngines([
        '.volt' => function ($view, $di) {
            $config = $this->getConfig();

            $volt = new VoltEngine($view, $di);

            $volt->setOptions([
                'compiledPath' => $config->application->cacheDir,
                'compileAlways' => $config->dev_environment,
                'stat' => $config->dev_environment,
                'compiledSeparator' => '_'
            ]);

            return $volt;
        },
        '.phtml' => 'Phalcon\Mvc\View\Engine\Php'
    ]);

return $view;

});

```
route micro:

``` php
$app->get('/', function () use ($app) {

    echo $app->view->render('system/index');

})->setName('home');
```

test volt:  system/index.volt

``` volt
<!doctype html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <title>Welcome!</title>
</head>
<body>
{%- macro print_hello(hello) %}
    {{ hello }}
{%- endmacro %}
{% print_hello('hi! baby.') %}
</body>
</html>

```
result:

```
Syntax error, unexpected token IDENTIFIER(print_hello) in /home/xerron/Repo/php/u-w-u.com/app/views/sui/system/index.volt on line 11

#0 [internal function]: Phalcon\Mvc\View\Engine\Volt\Compiler->_compileSource('...', false)
#1 [internal function]: Phalcon\Mvc\View\Engine\Volt\Compiler->compileFile('/home/xerron/Re...', '/home/xerron/Re...', false)
#2 [internal function]: Phalcon\Mvc\View\Engine\Volt\Compiler->compile('/home/xerron/Re...')
#3 [internal function]: Phalcon\Mvc\View\Engine\Volt->render('/home/xerron/Re...', Array, true)
#4 [internal function]: Phalcon\Mvc\View\Simple->_internalRender('system/index', Array)
#5 /home/xerron/Repo/php/u-w-u.com/app/modules/system/app.php(20): Phalcon\Mvc\View\Simple->render('system/index')
#6 [internal function]: Closure->{closure}()
#7 /home/xerron/Repo/php/u-w-u.com/web/index.php(78): Phalcon\Mvc\Micro->handle()
#8 {main}
```

**Comment by sergeyklay** [comment]
Could you please provide macros definition and the way in which you include it

**Comment by andresgutierrez** [comment]
This statement is wrong:

```volt
{% print_hello('hi! baby.') %} 
```

If you want to call the macro you have to you use `do`:

```volt
{% do print_hello('hi! baby.') %} 
```

or 

```volt
{{ print_hello('hi! baby.') }} 
```

**Comment by xerron** [comment]
Yes bug!.
I am try. 
``` twig
{{ print_hello('hi! baby.') }} 
```
and notice:

```
Notice: Access to undefined property _macros in /home/xerron/Repo/php/u-w-u.com/var/cache/default/volt/_home_xerron_repo_php_u-w-u.com_app_views_sui_user_register.volt.php on line 43
```

**Comment by Jurigag** [comment]
I thought it was fixed - https://github.com/phalcon/cphalcon/issues/11902 You sure you on 3.0.0 ?

**Comment by xerron** [comment]
Yes. Fix in branch 3.0.x. tank you. +1
